app/models/ems_refresh/parsers/azure.rb

```diff
@@ -0,0 +1,214 @@
+module EmsRefresh
+  module Parsers
+    class Azure < ManageIQ::Providers::CloudManager::RefreshParser
+      VALID_LOCATION = /\w+/i
```

Comment:
The i is not needed, since you don't have anything case sensitive in the pattern.